app/controllers/expenses_controller.rb

```diff
@@ -0,0 +1,107 @@
+class ExpensesController < ApplicationController
+  load_and_authorize_resource
+  skip_authorize_resource only: :index
+
+  before_action :set_expense, only: [:show, :edit, :update, :destroy, :hand_in, :update_item_count]
+  before_action :set_trip, only: [:new, :create, :update_item_count]
+
+  rescue_from CanCan::AccessDenied do |_exception|
+    flash[:error] = I18n.t('not_authorized')
+    redirect_to trips_path
+  end
+
+  def index
+    @expenses = Expense.where(user: current_user)
+  end
+
+  def show
+  end
+
+  def new
+    @trip = Trip.find(params[:trip_id])
```

Comment:
Du machst ja als before_action :set_trip. Das macht das gleiche wie in dieser Zeile hier. Die brauchst du also nicht.